### PR TITLE
Add possibility to set another user agent for the API requests

### PR DESCRIPTION
**Pull request by yrezgui**
This allows to set a custom user agent for the API requests made to IBM Watson services.

##### Checklist
- [x] `npm test` passes (tip: `npm run autofix` can correct most style issues)
- [ ] documentation is changed or added

### CODE REVIEW

File: lib/requestwrapper.js [hunk 1]
```diff
@@ -153,7 +153,7 @@ function createRequest(parameters, _callback) {
   // Headers
   options.headers = extend({}, options.headers);
 
-  if (!isBrowser) {
+  if (!isBrowser && !options.headers['User-Agent']) {
```
Comment (germanattanasio) [inline lib/requestwrapper.js]: We should add the `User-Agent` that the user sends us but not replace the one we are using.

File: lib/requestwrapper.js [hunk 2]
```diff
@@ -154,7 +154,8 @@ function createRequest(parameters, _callback) {
   options.headers = extend({}, options.headers);
 
   if (!isBrowser) {
-    options.headers['User-Agent'] = pkg.name + '-nodejs-' + pkg.version;
+    var suffix = options.headers['User-Agent'] ? ' ' + options.headers['User-Agent'] : '';
+    options.headers['User-Agent'] = pkg.name + '-nodejs-' + pkg.version + suffix;
```
Comment (germanattanasio) [inline lib/requestwrapper.js]:
```js
var providedAgent = options.headers['User-Agent']  ? ( '; ' + options.headers['User-Agent'] ) : '';
options.headers['User-Agent'] = pkg.name + '-nodejs-' + pkg.version + providedAgent;
```

File: lib/requestwrapper.js [hunk 3]
```diff
@@ -153,8 +153,9 @@ function createRequest(parameters, _callback) {
   // Headers
   options.headers = extend({}, options.headers);
 
-  if (!isBrowser && !options.headers['User-Agent']) {
-    options.headers['User-Agent'] = pkg.name + '-nodejs-' + pkg.version;
+  if (!isBrowser) {
+    var suffix = options.headers['User-Agent'] ? ' ' + options.headers['User-Agent'] : '';
```
Comment (germanattanasio) [inline lib/requestwrapper.js]: you need a `;` to separate the different values

**Comment by yrezgui** [comment]
@germanattanasio Done 👍

**Comment by nfriedly** [comment]
I'm iffy about this for a few reasons.

First, @yrezgui: why do you want change the User-Agent header?

Second, it already works if you use the "normalized" (lowercased) version (`options.headers['user-agent']`).

Finally @germanattanasio & @yrezgui: I'm not sure about append vs overwrite, but if we're going with append, then it should have a different parameter name, maybe `userAgentSuffix` or something.

**Comment by nfriedly** [comment]
I'm going to close this for now, we can come back to it later if we need to.

**Pull request closed without merging**